File: Python/shortestPalindrome.py
class Solution:
    def shortestPalindrome(self, s: str) -> str:
        r = s[::-1]
        for i in range(len(s)+1):
            print(r[i:])
            if s.startswith(r[i:]):
                print("End: ", r[:i])
                return r[:i]+s

File: Java/shortestPalindrome.java
class Solution {
    public String shortestPalindrome(String s) {
        String r = new StringBuilder(s).reverse().toString();
        for (int i = 0; i <= s.length(); i++) {
            String suffix = r.substring(i);
            if (s.startsWith(suffix)) {
                return r.substring(0, i) + s;
            }
        }
        return "";
    }
}